ingestion/src/metadata/utils/helpers.py

```diff
@@ -178,3 +191,60 @@ def datetime_to_ts(date: datetime) -> int:
     Convert a given date to a timestamp as an Int
     """
     return int(date.timestamp())
+
+
+def create_lineage(from_table, to_table, query_info):
+    try:
+        from_fqdn = f"{config.service_name}.{from_table}"
+        from_entity = metadata.get_by_name(entity=Table, fqdn=from_fqdn)
+        to_fqdn = f"{config.service_name}.{to_table}"
+        to_entity = metadata.get_by_name(entity=Table, fqdn=to_fqdn)
+        if not from_entity or not to_entity:
+            return None
+        from_entity_id = (
+            from_entity.id.__root__
+            if from_entity and hasattr(from_entity.id, "__root__")
+            else from_entity.id
+        )
+
+        to_entity_id = (
+            to_entity.id.__root__ if hasattr(to_entity.id, "__root__") else to_entity.id
+        )
+
+        lineage = AddLineageRequest(
+            edge=EntitiesEdge(
+                fromEntity=EntityReference(
+                    id=from_entity_id,
+                    type=query_info["from_type"],
+                ),
+                toEntity=EntityReference(
+                    id=to_entity_id,
+                    type=query_info["to_type"],
+                ),
+            )
+        )
+
+        created_lineage = metadata.add_lineage(lineage)
+        logger.info(f"Successfully added Lineage {created_lineage}")
+
+    except Exception as err:
+        logger.debug(traceback.print_exc())
+        logger.error(err)
+
+
+def ingest_lineage(query_info, metadata_config, config_obj):
+    global config
+    global metadata
```

Comment:
hi @codingwithabhi, what is the goal of playing with `global` here? Modifying a state like this can make things hard to debug in the future.

As a rule of thumb for easier maintenance, is that if we need any ingredients to run our functions, those should be passed as an argument.

If you need the `metadata` variable in `create_lineage`, it would be easier to pass that directly.

Please, let me know if I am missing something here